set/set-test.js

```diff
@@ -0,0 +1,70 @@
+"use strict";
+
+var QUnit = require("steal-qunit");
+
+var set  = require("./set");
+var canReflect = require("can-reflect");
+
+QUnit.module("can-key/set");
+
+QUnit.test("set single root", function (assert) {
+	var root = {
+		foo: "bar"
+	};
+	set(root, "foo", "baz");
+	assert.equal(root.foo, "baz", "got 'baz'");
+});
+
+QUnit.test("set deep objects", function (assert) {
+	var root = {
+		foo: {
+			bar: "baz"
+		}
+	};
+	set(root, "foo.bar", "new");
+	assert.equal(root.foo.bar, "new", "got 'new'");
+});
+
+QUnit.test("set with numeric index", function (assert) {
+	var list = [1, 2, 3];
+	set(list, 0, "one");
+
+	assert.equal(list[0], "one", "set the 1st element of the list");
+
+	set(list, 1, "two");
+	assert.equal(list[1], "two", "set the 2nd element of the list");
+});
+
+QUnit.test("set on an object that does not exist", function (assert) {
+	var root = {
+		foo: {}
+	};
+	set(root, "foo.bar.baz", "new");
+	assert.equal(root.foo.bar, undefined, "original object was not modified");
```

Comment:
What should happen in this case? Right now if one of the layers doesn’t exist, we just don’t set anything… maybe we should use the return value of `set()` to indicate whether something was set instead?